src/contracts/examples/fastlane-online/OuterHelpers.sol

```diff
@@ -43,6 +43,23 @@ contract OuterHelpers is FastLaneOnlineInner {
     //              CONTROL-LOCAL FUNCTIONS                //
     //                 (not delegated)                     //
     /////////////////////////////////////////////////////////
+
+    function setWinningSolver(address winningSolver) external {
+        // Only valid time this can be called is during the PostOps phase of a FLOnline metacall. When a user initiates
+        // that metacall with `fastOnlineSwap()` they are set as the user lock address. So the only time the check below
+        // will pass is when the caller of this function is the Execution Environment created for the currently active
+        // user and the FLOnline DAppControl.
+
+        (address expectedCaller,,) = IAtlas(ATLAS).getExecutionEnvironment(_getUserLock(), CONTROL);
+        if (msg.sender == expectedCaller) {
+            // Set winning solver in transient storage, to be used in `_updateSolverReputation()`
+            _setWinningSolver(winningSolver);
```

Comment:
I don't believe we need to add a check to make sure that winningSolver hasn't already been set because of Atlas's own locking mechanisms, but we may want to 'future proof' the FLO module for a recursive Atlas implementation by doing that here.  

The future vector would be Atlas metacall -> app -> atlas metacall

Reply:
I think that makes sense. Quite hard to reason about the security model (of Atlas and FLO) when nested metacalls are allowed. But I think this works and does not affect the current security model as Atlas/FLO are now: https://github.com/FastLane-Labs/atlas/pull/400/commits/765cf1834aa70bfb81e2b24f9e744db501891e1f